packages/backend/src/database/pagination/index.ts

```diff
@@ -0,0 +1,42 @@
+import { KnexPaginateArgs, KnexPaginatedData } from '@lightdash/common';
+import { Knex } from 'knex';
+import { KnexPaginationError } from './errors';
+
+export default class KnexPaginate {
+    static async paginate<T extends Knex.QueryBuilder>(
+        query: T,
+        paginateArgs?: KnexPaginateArgs,
+    ): Promise<KnexPaginatedData<typeof query>> {
+        if (paginateArgs) {
+            const { page, pageSize } = paginateArgs;
+            if (page < 1) {
+                throw new KnexPaginationError('page should be greater than 0');
+            }
+
+            if (pageSize < 1) {
+                throw new KnexPaginationError(
+                    'pageSize should be greater than 0',
+                );
+            }
+
+            const offset = (page - 1) * pageSize;
+            const count = await query.clone().clear('select').count();
```

Comment:
this is making an extra query to the database to get the number of rsults, right ? 

Should we use `Promises.all` ?  here ?

Reply:
Yeah, to calculate the total number of pages
Good point, we can add that here yeah 🤔 shouldn't cause any issue